### PR TITLE
Changelog entry for v0.2.1

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

**Preview**: https://pygmt-git-changelog-v021.gmt.vercel.app/changes.html

For reference, the previous changelog is at #599.

Changelog made by following the instructions at https://github.com/GenericMappingTools/pygmt/blob/master/MAINTENANCE.md#updating-the-changelog. The changelog itself includes the following sections:

- Highlights
- New features
- Enhancements
- Documentation
- Bug fixes
- Maintenance
- List of Contributors (sorted by number of commits)

TODO: 

- [x] Format Pull Request links using ``sed -i.bak -E 's$\(#([0-9]*)\)$(`#\1 <https://github.com/GenericMappingTools/pygmt/pull/\1>`__)$g' changes.rst``
- [x] Add Zenodo DOI link once it's reserved.
- [x] Add a new entry in doc/_static/version_switch.js for documentation switcher
- [x] Update citation information https://github.com/GenericMappingTools/pygmt#citing-pygmt
- [x] Set release date

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Notes**

- You can write `/format` in the first line of a comment to lint the code automatically

### CODE REVIEW

**Comment by seisman** [comment]
@weiji14 I've drafted a changelog for v0.2.1 release. Please see if you want to make some changes.

File: doc/changes.rst [hunk 1]
```diff
@@ -1,6 +1,79 @@
 Changelog
 =========
 
+Release v0.2.1 (2020/XX/XX)
```
Comment (seisman) [inline doc/changes.rst:4]: Need to update the date.

File: doc/changes.rst [hunk 2]
```diff
@@ -1,6 +1,79 @@
...
+---------------------------
+
+.. image:: https://zenodo.org/badge/DOI/10.5281/zenodo.4025418.svg
+   :alt: Digital Object Identifier for the Zenodo archive
+   :target: https://doi.org/10.5281/zenodo.4025418
```
Comment (seisman) [inline doc/changes.rst:9]: Need to update the DOI.

File: doc/changes.rst [hunk 3]
```diff
@@ -1,6 +1,79 @@
...
+* Check and lint sphinx configuration file doc/conf.py (#630)
+* Improve Makefile to clean ``__pycache__`` directory recursively (#611)
+* Update release process and checklist template (#602)
+
+This release contains contributions from:
+
+* `Dongdong Tian <https://github.com/seisman>`__
+* `Wei Ji Leong <https://github.com/weiji14>`__
+* `Conor Bacon <https://github.com/hemmelig>`__
+* `carocamargo <https://github.com/carocamargo>`__
```
Comment (seisman) [inline doc/changes.rst:75]: @carocamargo Thanks again for your contribution to the PyGMT project. We're going to release PyGMT v0.2.1 soon. Please let us know if you want to use the github username or real name here.

File: doc/changes.rst (hunk 2 above)
Comment (weiji14) [reply doc/changes.rst:9]: Zenodo DOI has been reserved for PyGMT v0.2.1. https://doi.org/10.5281/zenodo.4253459 (will work once we publish the release).

```suggestion
.. image:: https://zenodo.org/badge/DOI/10.5281/zenodo.4253459.svg
   :alt: Digital Object Identifier for the Zenodo archive
   :target: https://doi.org/10.5281/zenodo.4253459
```

File: doc/changes.rst [hunk 4]
```diff
@@ -1,6 +1,79 @@
...
+---------------------------
+
+.. image:: https://zenodo.org/badge/DOI/10.5281/zenodo.4025418.svg
+   :alt: Digital Object Identifier for the Zenodo archive
+   :target: https://doi.org/10.5281/zenodo.4025418
+
+Highlights
+
+* 🎉 **Patch release with more tutorials and gallery examples!** 🎉
+* 📹 Add `Liam <https://github.com/liamtoney>`__'s `ROSES 2020 PyGMT talk <https://www.youtube.com/watch?v=SSIGJEe0BIk>`__ (#643)
```
Comment (weiji14) [inline doc/changes.rst:14]: This depends on whether we make a v0.2.2 release, but should we add a statement about Python and GMT version support going forward (#644)? I.e. something like:

- PyGMT v0.2.x will be the last version to support Python 3.6, future versions of PyGMT(v0.3.x and above) will only support Python 3.7-3.9+
- PyGMT v0.2.x will be the last version to support GMT 6.1.1, future versions of PyGMT (v0.3.x and above) will require GMT 6.2.0 as a minimum.

Or we can do it in v0.2.2 (scheduled for December)?

File: doc/changes.rst (hunk 3 above)
Comment (weiji14) [reply doc/changes.rst:75]: Btw, If you want your name to be on the author list at Zenodo (https://zenodo.org/record/4025418), please also open up a Pull Request to add your name to https://github.com/GenericMappingTools/pygmt/edit/master/AUTHORS.md (and provide us with your ORCID and affiliation). Also cc @hemmelig on this.

File: doc/changes.rst (hunk 4 above)
Comment (seisman) [reply doc/changes.rst:14]: > we can do it in v0.2.2 (scheduled for December)?

GMT 6.2 is most likely to be released in January 2021 (https://forum.generic-mapping-tools.org/t/gmt-community-zoom-meeting-thursday-oct-22-utc-1700-hst-0700/943/6). There are at least two months before that. v0.2.2 in December (before AGU?) sounds good for a patch release. We may even have v0.2.3 before the release of GMT 6.2, depending on the changes we have made.

Comment (weiji14) [reply doc/changes.rst:14]: Ok, let's not add this notice then until PyGMT v0.2.2. I think one more patch release for the PyGMT v0.2.x series before GMT 6.2.0 comes out is enough (unless another maintainer steps up to do the work). Personally I *really* can't spend too much time maintaining PyGMT for the next two months until my thesis is finished :sweat_smile:

File: doc/changes.rst (hunk 2 above)
Comment (weiji14) [reply doc/changes.rst]: Btw, the BibTeX needs to be updated with the new Zenodo DOI too.

File: doc/changes.rst (hunk 4 above)
Comment (seisman) [reply doc/changes.rst:14]: Thesis always has the highest priority. 😃

File: doc/changes.rst [hunk 5]
```diff
@@ -1,6 +1,79 @@
...
+
+New Features
+
+* Wrap plot3d (#471)
+* Wrap grdfilter (#616)
+
+Enhancements
+
```
Comment (seisman) [inline doc/changes.rst:22]:
```suggestion
* Allow np.object dtypes into virtualfile_from_vectors (#684)
```

File: doc/changes.rst [hunk 6]
```diff
@@ -1,6 +1,79 @@
...
+* Add common alias cores (x) for grdimage and other multi-threaded modules (#625)
+* Enable switching different versions of documentation (#621)
+* Add common alias transparency (-t) to all plotting functions (#614)
+
+Bug Fixes
+
+* Disallow passing arguments like -XNone to GMT (#639)
+
+Maintenance
+
```
Comment (seisman) [inline doc/changes.rst:58]:
```suggestion
* Support Python 3.9 (#689)
```

**Review by weiji14 — COMMENTED** [review]
@seisman, I think we can go on and finish off this PR and cut a release. Could you run the `sed` command and handle these two tasks in this PR too?

- [ ] Add a new entry in doc/_static/version_switch.js for documentation switcher
- [ ] Update citation information https://github.com/GenericMappingTools/pygmt#citing-pygmt

File: doc/changes.rst [hunk 7]
```diff
@@ -1,6 +1,79 @@
...
+.. image:: https://zenodo.org/badge/DOI/10.5281/zenodo.4253459.svg
+   :alt: Digital Object Identifier for the Zenodo archive
+   :target: https://doi.org/10.5281/zenodo.4253459
+
+Highlights
+
+* 🎉 **Patch release with more tutorials and gallery examples!** 🎉
+* 📹 Add `Liam <https://github.com/liamtoney>`__'s `ROSES 2020 PyGMT talk <https://www.youtube.com/watch?v=SSIGJEe0BIk>`__ (#643)
+
```
[the remark this answers is not in the document]
Comment (seisman) [reply doc/changes.rst:16]: Done.

File: doc/changes.rst [hunk 8]
```diff
@@ -1,6 +1,81 @@
...
+Enhancements
+
+* Allow np.object dtypes into virtualfile_from_vectors (`#684 <https://github.com/GenericMappingTools/pygmt/pull/684>`__)
+* Let plot() accept record-by-record transparency (`#626 <https://github.com/GenericMappingTools/pygmt/pull/626>`__)
+* Refactor info to allow datetime inputs from xarray.Dataset and pandas.DataFrame tables (`#619 <https://github.com/GenericMappingTools/pygmt/pull/619>`__)
+
+Tutorials & Gallery
+
+* Add tutorial for pygmt.config (`#482 <https://github.com/GenericMappingTools/pygmt/pull/482>`__)
+* Add an example for different line styles (#604, #664)
```
Comment (weiji14) [inline doc/changes.rst:31]:
```suggestion
* Add an example for different line styles (`#604 <https://github.com/GenericMappingTools/pygmt/pull/604>`__, `#664 <https://github.com/GenericMappingTools/pygmt/pull/664>`__)
```

File: doc/changes.rst (hunk 1 above)
Comment (weiji14) [reply doc/changes.rst:4]:
```suggestion
Release v0.2.1 (2020/11/14)
```

File: doc/changes.rst (hunk 8 above)
Comment (seisman) [reply doc/changes.rst]: Nice catch. I believe it also means we should update our `sed` command.

File: doc/changes.rst [hunk 9]
```diff
@@ -1,6 +1,81 @@
...
+---------------------------
+
+.. image:: https://zenodo.org/badge/DOI/10.5281/zenodo.4253459.svg
+   :alt: Digital Object Identifier for the Zenodo archive
+   :target: https://doi.org/10.5281/zenodo.4253459
+
+Highlights
+
+* 🎉 **Patch release with more tutorials and gallery examples!** 🎉
+* Support Python 3.9 (`#689 <https://github.com/GenericMappingTools/pygmt/pull/689>`__)
```
Comment (weiji14) [inline doc/changes.rst:14]:
```suggestion
* 🐍 Support Python 3.9 (`#689 <https://github.com/GenericMappingTools/pygmt/pull/689>`__)
```
Use a snake emoji here?

Comment (seisman) [reply doc/changes.rst]: Great! Tried to search for "Python" but failed. :snake: looks good.

**Review by weiji14 — APPROVED** [review]

**Comment by seisman** [comment]
FYI, I also updated the release draft information at https://github.com/GenericMappingTools/pygmt/releases/tag/untagged-835457ed8c763a4fb621.

**Pull request merged**